tap_google_analytics/ga_client.py

```diff
@@ -240,23 +255,25 @@ def query_api(self, report_definition, pageToken=None):
         return self.analytics.reports().batchGet(
             body={
                 'reportRequests': [
-                {
-                    'viewId': self.view_id,
-                    'dateRanges': [{'startDate': self.start_date, 'endDate': self.end_date}],
-                    'pageSize': '1000',
-                    'pageToken': pageToken,
-                    'metrics': report_definition['metrics'],
-                    'dimensions': report_definition['dimensions'],
-                }]
+                    {
+                        'viewId': self.view_id,
+                        'dateRanges': [{'startDate': start_date, 'endDate': end_date}],
+                        'pageSize': '1000',
+                        'pageToken': pageToken,
+                        'metrics': report_definition['metrics'],
+                        'dimensions': report_definition['dimensions'],
+                    }]
             },
             quotaUser=self.quota_user
         ).execute()
 
-    def process_response(self, response):
+    def process_response(self, response, start_date, end_date):
```

Comment:
You set the function arguments `start_date, end_date` as required (no default values), are you sure that wherever this function is called, you passed new arguments?